server/src/main/java/org/elasticsearch/transport/Transport.java

```diff
@@ -134,7 +134,7 @@ default Version getVersion() {
          * Returns the version of the data to communicate in this channel.
          */
         default TransportVersion getTransportVersion() {
-            return getVersion().transportVersion;
+            return TransportVersion.CURRENT;
```

Comment:
Nearly all the things that don't override this are mocks/test implementations that don't care

Reply:
I would rather not have a `default` implementation of this method so that subclasses are forced to at least think about whether they care or not.